src/main/java/net/ripe/rpki/rsyncit/rrdp/RrdpFetcher.java

```diff
@@ -332,9 +336,10 @@ private static void checkResult(String objectUri, ValidationResult result) {
      * This MAY help for the corner case of objects having second-accuracy timestamps
      * and the timestatmp in seconds being the same for multiple objects.
      */
-    private Instant spiceWithHash(Instant t, byte[] hash) {
-        final BigInteger ms = new BigInteger(hash).mod(BigInteger.valueOf(1000L));
-        return t.truncatedTo(ChronoUnit.SECONDS).plusMillis(ms.longValue());
+    @VisibleForTesting
+    public static Instant incorporateHashInTimestamp(Instant t, byte[] hash) {
+        final BigInteger ms = new BigInteger(hash).mod(BigInteger.valueOf(1000_000_000L));
```

Comment:
Use `new BigInteger(1, hash)` to always interpret the value as a non-negative number? Also, constant is somewhat strangely formatted. Use `1_000_000_000L` instead?

Reply:
`.mod` will make it non-negative